### PR TITLE
Fix error on duplicate commonjs exports

### PR DESCRIPTION
**Pull request by sandersn**
Previously, the code missed setting the parent pointer for the lhs access expression.
Note that this was not previously an error; you can also avoid the error by setting the exclude flags back to `None`. But I don't think duplicate exports are that common, and non-checkjs users aren't going to see the error anyway.

Also add declaration emit of element access expressions, missed in my previous PR.

Fixes #40228

### CODE REVIEW

File: tests/baselines/reference/moduleExportAliasElementAccessExpression.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+//// [moduleExportAliasElementAccessExpression.js]
+function D () { }
+exports["D"] = D;
+
+
+//// [moduleExportAliasElementAccessExpression.js]
+function D() { }
+exports["D"] = D;
```
Comment (weswigham) [inline tests/baselines/reference/moduleExportAliasElementAccessExpression.js:8]: And non-identifier names, like `"A B"`, emit as...what? Probably needs a test.

**Review by DanielRosenwasser — APPROVED** [review]

File: tests/baselines/reference/moduleExportDuplicateAlias.errors.txt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+==== tests/cases/conformance/salsa/moduleExportAliasDuplicateAlias.js (3 errors) ====
+    exports.apply = undefined;
+            ~~~~~
+!!! error TS2300: Duplicate identifier 'apply'.
+    function a() { }
+    exports.apply = a;
+    ~~~~~~~~~~~~~
+!!! error TS2322: Type '() => void' is not assignable to type 'undefined'.
+            ~~~~~
+!!! error TS2300: Duplicate identifier 'apply'.
```
Comment (weswigham) [inline tests/baselines/reference/moduleExportDuplicateAlias.errors.txt:15]: To get rid of this error, we can defer detection of this case in the binder by removing the offending case from `AliasExcludes` (like we did for class/function merges) and then look for it in the _checker_, where we can filter certain declarations (probably undefined assignments) and except it, while retaining the error more generally.

I think this pattern is actually _super common_, considering it's our current emit for cjs, so we should probably invest the effort into making it work smoothly.

File: tests/baselines/reference/moduleExportAliasElementAccessExpression.js (hunk 1 above)
Comment (sandersn) [reply tests/baselines/reference/moduleExportAliasElementAccessExpression.js:8]: _A_B. I added a test the in the bug, but I guess I can add it here.

File: tests/baselines/reference/moduleExportDuplicateAlias.errors.txt (hunk 2 above)
Comment (sandersn) [reply tests/baselines/reference/moduleExportDuplicateAlias.errors.txt]: Per our discussion, I switched this to None for the beta. It turns out that the checker already issues an error in this case, but we can add an exclusion for `undefined`-as-initialiser later.

**Review by weswigham — APPROVED** [review]
I guess this is ok, so long as we're tracking fixing the error more broadly and adding an exception to the checker's duplicate export checking for `undefined` assignments (because our own emit looks like that). At least for the former, as I mentioned, I am still sliiiiiiightly worried that not using the appropriate exclude flags might result in some order-dependent cross-file merge issues down the line; so fixing it by adjusting the exclude flags themselves may be pertinent in the near future.

**Comment by sandersn** [comment]
Filed #40555 to follow up.

**Pull request merged**